src/i18n/number.js

```diff
@@ -7,7 +7,7 @@ const currencyDict = {
     'U.S. Dollar': 'USD'
 }
 
-export function formatCurrency(value, currency) {
+export function formatCurrency(value, currency, currencyDisplay = 'symbol') {
```

Comment:
can we validate that the display is either:

code, name, none or symbol and raise an error if not?

Or find a way that you do not need to know those values?

Maybe something similar to this:

export function getCurrencyFormatWithIsoCode(currencyCode) {
  return {
    style: 'currency',
    currency: currencyCode,
    currencyDisplay: 'code',
  }
}

export function getCurrencyFormatWithLocalName(currencyCode) {
  return {
    style: 'currency',
    currency: currencyCode,
    currencyDisplay: 'name',
  }
}

export function getCurrencyFormatNumbersOnly(currencyCode) {
  return {
    style: 'currency',
    currency: currencyCode,
    currencyDisplay: 'none',
  }
}

export function formatCurrency (value, format, lang) {
  const stripSymbols = (format.currencyDisplay === 'none')
  const localFormat = stripSymbols ? {...format, currencyDisplay: 'code'} : format
  let result = Intl.NumberFormat(lang, localFormat).format(value)
  if (stripSymbols) {
    result = result.replace(/[a-z]{3}/i, "").trim()
  }
  return result
}


But I am not sure I like the format thingy....

Reply:
Added type check which will throw error when valus is invalid